### PR TITLE
add headless web tests to `npm test`

### PR DESCRIPTION
**Pull request by smalls**
And ideally to Travis as well.

https://github.com/PolymerLabs/arcs/issues/454

### CODE REVIEW

**Comment by smalls** [comment]
It works!!

.. in that it fails the same way via travis as it does on my localhost, but the failure is far enough along that it's started chrome and begun interacting with Arcs.

https://travis-ci.org/PolymerLabs/arcs-cdn/builds/319880930?utm_source=github_status&utm_medium=notification

cantwontstop

**Comment by smalls** [comment]
Comment to rerun travis.

File: dev/test/wdio.conf.js [hunk 1]
```diff
@@ -1,4 +1,6 @@
 exports.config = {
+  port: '9515',
```
Comment (shaper) [inline dev/test/wdio.conf.js:2]: Is this standard webdriver port, or how did we choose this? Thought worth a comment perhaps.

File: dev/test/specs/starter-test.js [hunk 2]
```diff
@@ -78,12 +78,25 @@ function loadSeleniumUtils(cdnBranch) {
     script.src = `${cdnBranch}/test/selenium-utils.js`;
     document.getElementsByTagName('head')[0].appendChild(script);
   }, cdnBranch);
+  browser.waitUntil(() => {
+    try {
+      const result = browser.execute('pierceShadows(["head"])');
```
Comment (shaper) [inline dev/test/specs/starter-test.js:83]: Should we check result for an error or whatever expected value might be? Also, why do we pierce looking for `head` element, is that just arbitrary and we use success as a sign that `pierceShadows()` has loaded? If so also seems worth a comment.

File: dev/test/specs/starter-test.js [hunk 3]
```diff
@@ -78,12 +78,25 @@ function loadSeleniumUtils(cdnBranch) {
...
+    } catch (e) {
+      console.log(
+        `caught an exception looking for pierceShadows(); we'll try again. Text: ${e}`
```
Comment (shaper) [inline dev/test/specs/starter-test.js:86]: How many times do we try again? Is the retry handled by browser.waitUntil? Presumably at some point we'll stop trying so I wondered whether at that point this message would read incorrectly.

**Review by shaper — APPROVED** [review]

**Comment by smalls** [comment]
And there's a happy green dot on the test!

screen shot 2017-12-21 at 3 58 13 pm

**Pull request merged**